File: server/chainup_agent/api/schemas/skill_operation_spec.py
"""Admin / internal skill operation spec schemas."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class SkillOperationSpecSummaryOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_id: str = Field(alias="skillId")
    skill_spec_version: str = Field(alias="skillSpecVersion")
    lifecycle: str
    scenario_ids: list[str] = Field(default_factory=list, alias="scenarioIds")
    contract_complete: bool = Field(alias="contractComplete")
    spec_digest: str | None = Field(default=None, alias="specDigest")
    published_at: str | None = Field(default=None, alias="publishedAt")


class SkillOperationSpecListOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    items: list[SkillOperationSpecSummaryOut]


class SkillOperationSpecBodyOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_id: str = Field(alias="skillId")
    skill_spec_version: str = Field(alias="skillSpecVersion")
    body_markdown: str = Field(alias="bodyMarkdown")
    spec_digest: str | None = Field(default=None, alias="specDigest")
    source_git_ref: str | None = Field(default=None, alias="sourceGitRef")


class SkillSpecVersionEntryOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_spec_version: str = Field(alias="skillSpecVersion")
    lifecycle: str
    published_at: str | None = Field(default=None, alias="publishedAt")


class SkillSpecVersionHistoryOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_id: str = Field(alias="skillId")
    items: list[SkillSpecVersionEntryOut]


class SkillSpecPublishRequestIn(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_spec_version: str = Field(alias="skillSpecVersion")
    body_markdown: str | None = Field(default=None, alias="bodyMarkdown")
    spec_digest: str | None = Field(default=None, alias="specDigest")
    source_git_ref: str | None = Field(default=None, alias="sourceGitRef")


class SkillSpecPublishResultOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_id: str = Field(alias="skillId")
    skill_spec_version: str = Field(alias="skillSpecVersion")
    lifecycle: str
    spec_digest: str | None = Field(default=None, alias="specDigest")


class EffectiveSkillBodyOut(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    skill_id: str = Field(alias="skillId")
    skill_spec_version: str = Field(alias="skillSpecVersion")
    body_markdown: str = Field(alias="bodyMarkdown")
    spec_digest: str | None = Field(default=None, alias="specDigest")
    etag: str | None = None

File: product-doc/src/admin/src/api/types/skillOperationSpec.ts
/** 与 `specs/openapi/components/skill-operation-spec-schemas.yaml` 同窗 */

export type SkillSpecLifecycle = "DRAFT" | "PUBLISHED" | "DEPRECATED";

export type SkillOperationSpecSummary = {
  skillId: string;
  skillSpecVersion: string;
  lifecycle: SkillSpecLifecycle;
  scenarioIds?: string[];
  contractComplete?: boolean;
  specDigest?: string;
  publishedAt?: string | null;
};

export type SkillOperationSpecList = {
  items: SkillOperationSpecSummary[];
};

export type SkillOperationSpecBody = {
  skillId: string;
  skillSpecVersion: string;
  bodyMarkdown: string;
  specDigest?: string;
  sourceGitRef?: string | null;
};

export type SkillSpecPublishRequest = {
  skillSpecVersion: string;
  bodyMarkdown?: string;
  sourceGitRef?: string | null;
  specDigest?: string;
};

export type SkillSpecPublishResult = {
  skillId: string;
  skillSpecVersion: string;
  lifecycle: SkillSpecLifecycle;
  specDigest?: string;
};

export type EffectiveSkillSpecResponse = {
  skillId: string;
  skillSpecVersion: string;
  bodyMarkdown: string;
  specDigest?: string;
  etag?: string;
};

export type SkillSpecVersionHistory = {
  skillId: string;
  items: Array<{
    skillSpecVersion: string;
    lifecycle: SkillSpecLifecycle;
    publishedAt?: string;
  }>;
};

export type SkillPublishProblem = {
  type?: string;
  title?: string;
  detail?: string;
  code?: "PROMPT_SKILL_REF_INVALID" | "SKILL_PUBLISH_GATE_FAILED" | "SKILL_VERSION_NOT_MONOTONIC";
};
